logic/src/commonTest/kotlin/com/wire/kalium/logic/feature/featureConfig/GetFileSharingStatusUseCaseTest.kt

```diff
@@ -0,0 +1,107 @@
+import com.wire.kalium.logic.data.featureConfig.FeatureConfigRepository
+import com.wire.kalium.logic.data.featureConfig.FileSharingModel
+import com.wire.kalium.logic.data.message.MessageRepository
+import com.wire.kalium.logic.data.user.ConnectionState
+import com.wire.kalium.logic.data.user.SelfUser
+import com.wire.kalium.logic.data.user.UserRepository
+import com.wire.kalium.logic.feature.asset.SendImageMessageResult
+import com.wire.kalium.logic.feature.asset.SendImageMessageUseCaseImpl
+import com.wire.kalium.logic.feature.asset.SendImageUseCaseTest
+import com.wire.kalium.logic.feature.client.SelfClientsResult
+import com.wire.kalium.logic.feature.message.MessageSender
+import com.wire.kalium.logic.feature.session.PushTokenUseCaseTest
+import com.wire.kalium.logic.functional.Either
+import com.wire.kalium.logic.test_util.TestNetworkException
+import com.wire.kalium.network.api.user.pushToken.PushTokenBody
+import com.wire.kalium.network.exceptions.KaliumException
+import io.mockative.Mock
+import io.mockative.any
+import io.mockative.classOf
+import io.mockative.given
+import io.mockative.mock
+import io.mockative.once
+import io.mockative.verify
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.flow.flowOf
+import kotlinx.coroutines.test.runTest
+import kotlin.test.BeforeTest
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertIs
+
+@ExperimentalCoroutinesApi
+class GetFileSharingStatusUseCaseTest {
+
+
+    @Test
+    fun givenRepositoryCallIsSuccessful_thenSuccessIsReturned() = runTest {
+        val (arrange, getFileSharingStatusUseCase) = Arrangement()
+            .withSuccessfulResponse()
+            .arrange()
+
+        val actual = getFileSharingStatusUseCase.invoke()
+        assertIs<GetFileSharingStatusResult.Success>(actual)
+        assertEquals(arrange.fileSharingModel, actual.fileSharingModel)
+        verify(arrange.featureConfigRepository)
+            .coroutine { getFileSharingFeatureConfig() }
+            .wasInvoked(exactly = once)
+    }
+
+
+    @Test
+    fun givenRepositoryCallFailWithInvalidCredentials_thenOperationDeniedIsReturned() = runTest {
+        val (arrange, getFileSharingStatusUseCase) = Arrangement()
+            .withErrorResponse()
+            .arrange()
+
+        val actual = getFileSharingStatusUseCase.invoke()
+
+        assertIs<GetFileSharingStatusResult.Failure.OperationDenied>(actual)
+        assertEquals(arrange.operationDeniedFail, actual)
+
+        verify(arrange.featureConfigRepository)
+            .coroutine { getFileSharingFeatureConfig() }
+            .wasInvoked(exactly = once)
+    }
+
+
+    private class Arrangement {
+
+        val fileSharingModel = FileSharingModel("locked", "enabled")
+        val expectedFail = NetworkFailure.ServerMiscommunication(TestNetworkException.invalidCredentials)
+        val operationDeniedFail = GetFileSharingStatusResult.Failure.OperationDenied
```

Comment:
Ideally the expected responses should be provided by the test on the arrangement creation, although they could be defined on a private function. Otherwise you lose visibility on what's being returned. 

It might not be a problem now that the test suite is small, but could become a bit annoying once the feature grows and multiple calls to different repositories are done and several possible errors could be thrown